packages/services/src/simulated-trade-execution.service.ts

```diff
@@ -680,20 +682,24 @@ export class SimulatedTradeExecutionService {
         agentId,
         competitionId,
       );
-    // TODO: maxTradePercentage should probably be a setting per comp.
-    const maxTradeValue =
-      portfolioValue * (this.config.maxTradePercentage / 100);
+    // Get maxTradePercentage from database, fallback to config default
+    const paperTradingConfig =
+      await this.paperTradingConfigRepo.findByCompetitionId(competitionId);
```

Comment:
I think caching this might be a good idea, especially as we might have concurrent paper trading comps, each with peak trade volumes of possibly upwards of 100 trades/minute. Since `maxTradePercentage` is unlikely to change for a specific competition, we could reduce some db load here